powersimdata/input/change_table.py

```diff
@@ -789,17 +794,92 @@ def _get_df_with_new_elements(self, table):
             'branch', 'bus', 'dcline', or 'plant'.
         :return: (*pandas.DataFrame*) -- the post-transformation table.
         """
-        add_key = f"new_{table}"
-        if add_key not in self.ct:
+        modification_keys = [f"new_{table}"]
+        if table == "plant":
+            modification_keys += sorted(self.grid.plant["type"].unique())
+        if not any(m in self.ct for m in modification_keys):
             return getattr(self.grid, table)
-        new_elements_tuple = tuple(tuple(sorted(b.items())) for b in self.ct[add_key])
-        if new_elements_tuple in self._new_element_caches[table]:
-            return self._new_element_caches[table][new_elements_tuple]
+        modified_elements_tuple = tuple(
+            chain.from_iterable(
+                [
+                    f"new_{table}",
+                    tuple(tuple(sorted(b.items())) for b in self.ct.get(table, {})),
```

Comment:
We need the extra `tuple`? I know it's already there, just curious what the data looks like.. kinda thought it was idempotent.

Reply:
I think the outermost `tuple` in this line is needed, otherwise we end up with a generator instead of something that can be hashed (unless you're talking about the `tuple` call wrapping `chain.from_iterable`).

I realize how clunky this block is, if you come up with something cleaner I'll happily take it. The test case that I added should activate this block to validate any refactoring attempts.

Reply:
Makes sense, I misread the parenthesis..